support-frontend/assets/components/footerCompliant/containers/contentStyles.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import styled from '@emotion/styled';
+import { from } from '@guardian/src-foundations/mq';
+import { brand } from '@guardian/src-foundations/palette';
+import { space } from '@guardian/src-foundations';
+
+const marginRules = `display: block;
+content: "";
+flex-basis: 0;
+flex-grow: 1;`;
+
+type WithMarginsProps = {|
+  before: boolean,
+  after: boolean
+|}
+
+export const WithMargins = styled.div`
+  display: flex;
+  ${from.tablet} {
+    ${({ before }: WithMarginsProps) => (before ? `&:before { ${marginRules} }` : '')}
+    ${({ after }: WithMarginsProps) => (after ? `&:after { ${marginRules} }` : '')}
+  }
+`;
```

Comment:
In the past we have recommended not using `styled` to create or extend components. Every `styled` component rendered creates an additional `Context.Consumer` component, which can lead to [performance issues](https://calendar.perfplanet.com/2019/the-unseen-performance-costs-of-css-in-js-in-react-apps/). Things may have changed since the analysis in that article was performed. Also this will only become appararent if you use these components a significant number of times on your page (say, more than 50).

The alternative using `css` is more verbose:

```tsx
const withMarginsStyle = ({before, after}: WithMarginsProps) => css`
  display: flex;
  ${from.tablet} {
    ${before ? `&:before { ${marginRules} }` : '')}
    ${after ? `&:after { ${marginRules} }` : '')}
  }
`

export const WithMargins = ({children, ...props}) => (
  <div css={withMarginsStyle(props)}>
    {children}
  </div>
)
```

I do find this approach clearer, and more consistent with the rest of your components. The inline functions in the `styled` approach are a bit magical. That might just be a personal thing 😊

Reply:
Actually I just realised you can this more efficient and avoid the logic in the CSS altogether:

```tsx
const withMarginsStyle = css`
  display: flex;
`

const afterStyle = css`
  ${from.tablet} {
    &:after { 
      ${marginRules} 
    }
  }
`

const beforeStyle = css`
  ${from.tablet} {
    &:before { 
      ${marginRules} 
    }
  }
`

export const WithMargins = ({children, before, after}) => (
  <div css={[
     withMarginsStyle,
     before ? beforeStyle : '',
     after ? afterStyle : '',
   ]}>
    {children}
  </div>
)
```

Reply:
This is the most verbose, but avoiding dynamic styles is, in the long term, a worthwhile investment. If all styles can be made static, they can be identified and extracted at build time. This would avoid the user's device having to do the work of calculating which styles to apply, and may save us from needing a runtime like Emotion running on the client at all (I'm pushing [these docs for Treat JS](https://seek-oss.github.io/treat/background) on everyone I know right now!)

Reply:
That makes a lot of sense, thanks! And I agree on the benefits of minimising the work that has to get done on the device. I'll make some changes.